### MURDER GAME

In [7]:
import pandas as pd
import numpy as np
import sqlite3
import os

In [14]:
# Creamos la conexión para DB local

conn = sqlite3.connect("sql-murder-mystery.db")

# Creamos el comando o cursor
cursor = conn.cursor()

In [18]:
q = """
    SELECT description
    FROM crime_scene_report
    where type='murder' AND date="20180115" AND city="SQL City"
"""
df = pd.read_sql(q, conn)
df

description
0  Security footage shows that there were 2 witne...

In [21]:
q = """
    SELECT *
    FROM person
    where address_street_name ="Northwestern Dr"
    ORDER BY address_number DESC
    LIMIT(1)
"""
df = pd.read_sql(q, conn)
df

id            name  license_id  address_number address_street_name   
0  14887  Morty Schapiro      118009            4919     Northwestern Dr  \

         ssn  
0  111564949

In [22]:
q = """
    SELECT *
    FROM person
    where name LIKE "%Annabel%" and address_street_name="Franklin Ave"
    LIMIT(1)
"""
df = pd.read_sql(q, conn)
df

id            name  license_id  address_number address_street_name   
0  16371  Annabel Miller      490173             103        Franklin Ave  \

         ssn  
0  318771143

In [23]:
q = """
    SELECT *
    FROM interview
    where person_id=14887
    LIMIT(1)
"""
df = pd.read_sql(q, conn)
df

person_id                                         transcript
0      14887  I heard a gunshot and then saw a man run out. ...

* id	name	license_id	address_number	address_street_name	ssn  
0	14887	Morty Schapiro	118009	4919	Northwestern Dr	111564949

* I heard a gunshot and then saw a man run out. He had a "Get Fit Now Gym" bag. The membership number on the bag started with "48Z". Only gold members have those bags. The man got into a car with a plate that included "H42W".

In [24]:
q = """
    SELECT *
    FROM interview
    where person_id=16371
    LIMIT(1)
"""
df = pd.read_sql(q, conn)
df

person_id                                         transcript
0      16371  I saw the murder happen, and I recognized the ...

* id	name	license_id	address_number	address_street_name	ssn  
0	16371	Annabel Miller	490173	103	Franklin Ave	318771143

* I saw the murder happen, and I recognized the killer from my gym when I was working out last week on January the 9th.

In [48]:
q = """
    SELECT *
    FROM get_fit_now_member
    WHERE id LIKE "48Z%" AND membership_status="gold"
"""
df = pd.read_sql(q, conn)
df

id  person_id           name  membership_start_date membership_status
0  48Z7A      28819   Joe Germuska               20160305              gold
1  48Z55      67318  Jeremy Bowers               20160101              gold

In [41]:
q = """
 SELECT person.id, person.name
 FROM person
 JOIN get_fit_now_member
 ON person.id=get_fit_now_member.person_id
 JOIN drivers_license
 ON person.license_id=drivers_license.id
 WHERE drivers_license.id LIKE "48Z%" AND membership_status="gold" AND plate_number LIKE "%H42W%"
"""
df = pd.read_sql(q, conn)
df

Empty DataFrame
Columns: [id, name]
Index: []

In [52]:
q = """
 SELECT id, plate_number
 FROM drivers_license
 WHERE plate_number LIKE "%H42W%"
"""
df = pd.read_sql(q, conn)
df

id plate_number
0  183779       H42W0X
1  423327       0H42W2
2  664760       4H42WR

In [50]:
q = """
SELECT license_id 
   from person
   WHERE id = 67318  or id = 28819
   """
df = pd.read_sql(q, conn)
df

license_id
0      173289
1      423327

In [51]:
q = """
   SELECT plate_number 
   from drivers_license
   WHERE id = 173289  or id = 423327
   """
df = pd.read_sql(q, conn)
df

plate_number
0       0H42W2

In [53]:

q = """
   SELECT transcript 
   from interview
   WHERE person_id=67318
   """
df = pd.read_sql(q, conn)
df

transcript
0  I was hired by a woman with a lot of money. I ...

** I was hired by a woman with a lot of money. I don't know her name but I know she's around 5'5" (65") or 5'7" (67"). She has red hair and she drives a Tesla Model S. I know that she attended the SQL Symphony Concert 3 times in December 2017

In [ ]:
q = """
   SELECT transcript 
   from interview
   WHERE person_id=67318
   """
df = pd.read_sql(q, conn)
df

In [58]:
q = """
 SELECT person.id
 FROM person
 JOIN drivers_license
 ON person.license_id=drivers_license.id
 WHERE height>65 AND height<67 AND hair_color="red" AND car_make="Tesla" AND car_model="Model S"
   """
df = pd.read_sql(q, conn)
df

id
0  90700
1  99716

*SQL Symphony Concert 3 times in December 2017

In [62]:
q = """
WITH t1 AS (
 SELECT person.id, person.name
 FROM person
 JOIN drivers_license
 ON person.license_id=drivers_license.id
 WHERE height>65 AND height<67 AND hair_color="red" AND car_make="Tesla" AND car_model="Model S"
)
SELECT t1.id, t1.name
FROM t1
JOIN facebook_event_checkin as fb
ON t1.id=fb.person_id
WHERE event_name="SQL Symphony Concert" AND date LIKE "201712%"

   """
df = pd.read_sql(q, conn)
df

id              name
0  99716  Miranda Priestly
1  99716  Miranda Priestly
2  99716  Miranda Priestly

In [63]:
q = """
SELECT transcript
FROM interview
WHERE person_id=99716
   """
df = pd.read_sql(q, conn)
df

Empty DataFrame
Columns: [transcript]
Index: []

In [ ]:
q = """

   """
df = pd.read_sql(q, conn)
df